No message for badly styled __init__ method arguments

<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this, make sure that the issue doesn't already exist
  or if it is not closed.

  Is your issue fixed on the preview release?: pip install pylint astroid --pre -U

-->

### Steps to reproduce
1. create file `foo_bar.py`:
```python
class FooBar:
    def __init__(self, fooBar):
        self._foo_bar = fooBar

    def foo_bar(self, foo_bar_arg):
        print(self._foo_bar + foo_bar_arg)

    def bar_bar(self, foo_bar_arg):
        print(self._foo_bar + foo_bar_arg)
```
2. Run: `pylint foo_bar`

### Current behavior

No complaints about the `fooBar` argument (should be `snake_case`).

### Expected behavior

Issues a message that `fooBar` is not in `snake_case` style.

The following example issues correctly `foo_bar.py:6:4: C0103: Argument name "fooBarArg" doesn't conform to `snake_case` naming style (invalid-name)` for the `foo_bar` method, but not for the `__init__` method.

```python
class FooBar:
    def __init__(self, fooBar):
        self._foo_bar = fooBar

    def foo_bar(self, fooBarArg):
        print(self._foo_bar + fooBarArg)

    def bar_bar(self, foo_bar_arg):
        print(self._foo_bar + foo_bar_arg)
```

Edit: it does so with the default settings and with my custom settings with I have added as attachement.
[pylintrc.txt](https://github.com/PyCQA/pylint/files/4616408/pylintrc.txt)



### pylint --version output

```
pylint 2.5.2
astroid 2.4.1
Python 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -47,6 +47,10 @@ Release date: TBA
 
   Closes #5066
 
+* Fixed ``invalid-name`` not checking parameters of overwritten base ``object`` methods
+
+  Closes #3614
+
 
 What's New in Pylint 2.11.1?
 ============================

--- a/pylint/checkers/utils.py
+++ b/pylint/checkers/utils.py
@@ -456,8 +456,11 @@ def assign_parent(node: nodes.NodeNG) -> nodes.NodeNG:
 
 
 def overrides_a_method(class_node: nodes.ClassDef, name: str) -> bool:
-    """return True if <name> is a method overridden from an ancestor"""
+    """return True if <name> is a method overridden from an ancestor
+    which is not the base object class"""
     for ancestor in class_node.ancestors():
+        if ancestor.name == "object":
+            continue
         if name in ancestor and isinstance(ancestor[name], nodes.FunctionDef):
             return True
     return False

--- a/tests/functional/i/invalid/invalid_name.py
+++ b/tests/functional/i/invalid/invalid_name.py
@@ -1,5 +1,7 @@
 """ Tests for invalid-name checker. """
-# pylint: disable=unused-import,  wrong-import-position,import-outside-toplevel
+# pylint: disable=unused-import, wrong-import-position, import-outside-toplevel, missing-class-docstring
+# pylint: disable=too-few-public-methods
+
 
 AAA = 24
 try:
@@ -67,3 +69,8 @@ def a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad():  # Shoul
     print('LOL')
 
 a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad()
+
+
+class FooBar:
+    def __init__(self, fooBar) -> None: # [invalid-name]
+        self.foo_bar = fooBar

--- a/tests/functional/i/invalid/invalid_name.txt
+++ b/tests/functional/i/invalid/invalid_name.txt
@@ -1,5 +1,6 @@
-invalid-name:10:0::"Constant name ""aaa"" doesn't conform to UPPER_CASE naming style"
-invalid-name:14:4::"Constant name ""time"" doesn't conform to UPPER_CASE naming style"
-invalid-name:30:0:a:"Function name ""a"" doesn't conform to snake_case naming style"
-invalid-name:44:4::"Constant name ""Foocapfor"" doesn't conform to UPPER_CASE naming style"
-invalid-name:61:0:a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad:"Function name ""a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad"" doesn't conform to snake_case naming style"
+invalid-name:12:0::"Constant name ""aaa"" doesn't conform to UPPER_CASE naming style":HIGH
+invalid-name:16:4::"Constant name ""time"" doesn't conform to UPPER_CASE naming style":HIGH
+invalid-name:32:0:a:"Function name ""a"" doesn't conform to snake_case naming style":HIGH
+invalid-name:46:4::"Constant name ""Foocapfor"" doesn't conform to UPPER_CASE naming style":HIGH
+invalid-name:63:0:a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad:"Function name ""a_very_very_very_long_function_name_WithCamelCase_to_make_it_sad"" doesn't conform to snake_case naming style":HIGH
+invalid-name:75:4:FooBar.__init__:"Argument name ""fooBar"" doesn't conform to snake_case naming style":HIGH

